src/platform/FailSafeContext.cpp

```diff
@@ -65,9 +82,89 @@ CHIP_ERROR FailSafeContext::DisarmFailSafe()
     mFailSafeArmed                  = false;
     mAddNocCommandHasBeenInvoked    = false;
     mUpdateNocCommandHasBeenInvoked = false;
+    ConfigurationMgr().SetFailSafeArmed(false);
     DeviceLayer::SystemLayer().CancelTimer(HandleArmFailSafe, this);
+
+    if (DeleteFromStorage() != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to delete the captured FailSafeContext from persistent storage");
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void FailSafeContext::SetAddNocCommandInvoked(FabricIndex nocFabricIndex)
+{
+    mAddNocCommandHasBeenInvoked = true;
+    mFabricIndex                 = nocFabricIndex;
+
+    if (CommitToStorage() != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to commit the AddNocCommandHasBeenInvoked of FailSafeContext to persistent storage");
+    }
+}
+
+void FailSafeContext::SetUpdateNocCommandInvoked(FabricIndex nocFabricIndex)
+{
+    mUpdateNocCommandHasBeenInvoked = true;
+    mFabricIndex                    = nocFabricIndex;
+
+    if (CommitToStorage() != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to commit the UpdateNocCommandHasBeenInvoked of FailSafeContext to persistent storage");
+    }
+}
+
+CHIP_ERROR FailSafeContext::CommitToStorage()
+{
+    uint8_t buf[FailSafeContextTLVMaxSize()];
+    TLV::TLVWriter writer;
+    writer.Init(buf);
+
+    TLV::TLVType outerType;
+    ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outerType));
+    ReturnErrorOnFailure(writer.Put(kFabricIndexTag, mFabricIndex));
+    ReturnErrorOnFailure(writer.Put(kAddNocCommandTag, mAddNocCommandHasBeenInvoked));
+    ReturnErrorOnFailure(writer.Put(kUpdateNocCommandTag, mUpdateNocCommandHasBeenInvoked));
+    ReturnErrorOnFailure(writer.EndContainer(outerType));
+
+    const auto failSafeContextTLVLength = writer.GetLengthWritten();
+    VerifyOrReturnError(CanCastTo<uint16_t>(failSafeContextTLVLength), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    return PersistedStorage::KeyValueStoreMgr().Put(kFailSafeContextKey, buf, static_cast<uint16_t>(failSafeContextTLVLength));
```

Comment:
Please use the key allocator class we have instead of hardcoding a key in this file.